lib/velum/salt_api.rb

```diff
@@ -95,10 +95,8 @@ def perform_request(endpoint:, method:, data: {}, client: nil)
 
         opts = { use_ssl: false, open_timeout: 2 }
         Net::HTTP.start(uri.hostname, uri.port, opts) { |http| http.request(req) }
-      rescue *HTTP_EXCEPTIONS
-        # TODO: we might want to wrap the original exception inside the SaltConnectionException
-        # so we can show detailed problems to the user in the future.
-        raise SaltConnectionException
+      rescue *HTTP_EXCEPTIONS => e
+        raise SaltConnectionException, e
```

Comment:
It's OK. I was thinking something more in the lines of `raise SaltConnectionException.new(e)` and keep the original exception as an attribute of `SaltConnectionException`, but we can completely ignore this for now.